src/main/java/net/obvj/junit/utils/matchers/StringMatcher.java

```diff
@@ -0,0 +1,188 @@
+                        return false;
+                    }
+                }
+                return true;
+            }
+        },
+
+        /**
+         * Matches if any of the specified substrings is found in the tested string.
+         */
+        ANY
+        {
+            @Override
+            public boolean evaluate(String string, List<String> substrings, Description mismatch)
+            {
+                for (String substring : substrings)
+                {
+                    if (string.contains(substring))
+                    {
+                        return true;
+                    }
+                }
+                mismatch.appendText(String.format(NONE_OF_THE_STRINGS_FOUND, string));
+                return false;
+            }
+        },
+
+        /**
+         * Matches if none of the specified substrings is found in the tested string.
+         */
+        NONE
+        {
+            @Override
+            public boolean evaluate(String string, List<String> substrings, Description mismatch)
+            {
+                for (String substring : substrings)
+                {
+                    if (string.contains(substring))
+                    {
+                        mismatch.appendText(String.format(UNEXPECTED_STRING_FOUND, substring, string));
+                        return false;
+                    }
+                }
+                return true;
+            }
+        };
+
+        /**
+         * Execute the matcher business logic.
+         *
+         * @param string     the string to be evaluated
+         * @param substrings the substrings to be checked
+         * @param mismatch   the description to be used for reporting in case of mismatch
+         */
+        public abstract boolean evaluate(String string, List<String> substrings, Description mismatch);
+
+    }
+
+    private final Strategy strategy;
+    private final List<String> substrings;
+
+    /**
+     * Builds this Matcher with a specialized strategy and a list of substrings to be tested.
+     *
+     * @param strategy   the evaluation strategy
+     * @param substrings the substrings to be evaluated
+     */
+    private StringMatcher(Strategy strategy, String... substrings)
+    {
+        this.strategy = strategy;
+        this.substrings = Arrays.asList(substrings);
+    }
+
+    /**
+     * Creates a matcher that matches if the examined string contains <b>all</b> of the
+     * specified substrings (regardless of the order they appear in the string).
+     * <p/>
+     * For example:
+     *
+     * <pre>
+     * assertThat("the quick brown fox", containsAll("fox", "the"))
+     * </pre>
+     */
+    @Factory
+    public static Matcher<String> containsAll(String... substrings)
+    {
+        return new StringMatcher(Strategy.ALL, substrings);
+    }
+
+    /**
+     * Creates a matcher that matches if the examined string contains <b>any</b> of the
+     * specified substrings.
+     * <p/>
```

Comment:
self-closing element not allowed.
```suggestion
     * <p>
```